In [1]:
import pandas as pd
import numpy as np

In [2]:
movies_df = pd.read_csv('HW1-data.csv')

In [3]:
movies_df

User  Gender (1 =F, 0=M)  260: Star Wars: Episode IV - A New Hope (1977)  \
0    755                   0                                             1.0   
1   5277                   0                                             5.0   
2   1577                   1                                             NaN   
3   4388                   0                                             NaN   
4   1202                   1                                             4.0   
5   3823                   1                                             2.0   
6   5448                   0                                             NaN   
7   5347                   0                                             4.0   
8   4117                   1                                             5.0   
9   2765                   0                                             4.0   
10  5450                   1                                             2.0   
11   139                   0                                             3.0   
12  1940                   0                                             2.0   
13  3118                   1                                             3.0   
14  4656                   1                                             4.0   
15  4796                   1                                             NaN   
16  6037                   0                                             NaN   
17  3048                   1                                             4.0   
18  4790                   0                                             5.0   
19  4489                   0                                             1.0   

    1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
0                                                 5.0         
1                                                 3.0         
2                                                 NaN         
3                                                 3.0         
4                                                 3.0         
5                                                 4.0         
6                                                 NaN         
7                                                 NaN         
8                                                 1.0         
9                                                 2.0         
10                                                1.0         
11                                                5.0         
12                                                3.0         
13                                                NaN         
14                                                4.0         
15                                                NaN         
16                                                NaN         
17                                                5.0         
18                                                1.0         
19                                                2.0         

    356: Forrest Gump (1994)  318: Shawshank Redemption, The (1994)  \
0                        2.0                                    NaN   
1                        NaN                                    2.0   
2                        NaN                                    5.0   
3                        NaN                                    NaN   
4                        4.0                                    1.0   
5                        4.0                                    4.0   
6                        3.0                                    1.0   
7                        NaN                                    NaN   
8                        NaN                                    4.0   
9                        NaN                                    5.0   
10                       5.0                                    NaN   
11                       2.0                                    NaN   
12                       NaN                                    5.0   
13   

In [4]:
movies_df.shape

(20, 22)

#### Mean Rating: Calculate the mean rating for each movie, order with the highest rating listed first, and submit the top three (along with the mean scores for the top two).

In [5]:
mean_rating_df = movies_df.iloc[:,2:].T.mean(axis=1, skipna=True).sort_values(ascending=False)

In [6]:
mean_rating_df

318: Shawshank Redemption, The (1994)                      3.600000
260: Star Wars: Episode IV - A New Hope (1977)             3.266667
541: Blade Runner (1982)                                   3.222222
1265: Groundhog Day (1993)                                 3.166667
593: Silence of the Lambs, The (1991)                      3.062500
296: Pulp Fiction (1994)                                   3.000000
1210: Star Wars: Episode VI - Return of the Jedi (1983)    3.000000
2028: Saving Private Ryan (1998)                           3.000000
34: Babe (1995)                                            3.000000
527: Schindler's List (1993)                               3.000000
3578: Gladiator (2000)                                     2.916667
2396: Shakespeare in Love (1998)                           2.909091
1198: Raiders of the Lost Ark (1981)                       2.909091
2571: Matrix, The (1999)                                   2.833333
2762: Sixth Sense, The (1999)                   

#### Rating Count (popularity): 
Count the number of ratings for each movie, order with the most number of ratings first, and submit the top three (along with the counts for the top two).

In [7]:
ranking_df = movies_df.iloc[:,2:].T.count(axis=1).sort_values(ascending=False)

In [8]:
ranking_df

1: Toy Story (1995)                                        17
593: Silence of the Lambs, The (1991)                      16
260: Star Wars: Episode IV - A New Hope (1977)             15
1210: Star Wars: Episode VI - Return of the Jedi (1983)    14
780: Independence Day (ID4) (1996)                         13
2762: Sixth Sense, The (1999)                              12
527: Schindler's List (1993)                               12
2571: Matrix, The (1999)                                   12
1265: Groundhog Day (1993)                                 12
2916: Total Recall (1990)                                  12
3578: Gladiator (2000)                                     12
2028: Saving Private Ryan (1998)                           11
1259: Stand by Me (1986)                                   11
296: Pulp Fiction (1994)                                   11
1198: Raiders of the Lost Ark (1981)                       11
2396: Shakespeare in Love (1998)                           11
318: Sha

#### % of ratings 4+ (liking): 
Calculate the percentage of ratings for each movie that are 4 or higher.
Order with the highest percentage first, and submit the top three (along with the percentage for the top two). 
Notice that the three different measures of "best" reflect different priorities and give different results; 
this should help you see why you need to be thoughtful about what metrics you use.

In [9]:
ranking_perc_df = movies_df.iloc[:,2:].apply(lambda x: x.value_counts()).fillna(0)

In [10]:
ranking_perc_df

260: Star Wars: Episode IV - A New Hope (1977)  \
1.0                                               2   
2.0                                               3   
3.0                                               2   
4.0                                               5   
5.0                                               3   

     1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
1.0                                                  3         
2.0                                                  2         
3.0                                                  4         
4.0                                                  2         
5.0                                                  3         

     356: Forrest Gump (1994)  318: Shawshank Redemption, The (1994)  \
1.0                         2                                    2.0   
2.0                         3                                    1.0   
3.0                         2                                    0.0   
4.0                         2                                    3.0   
5.0                         1                                    4.0   

     593: Silence of the Lambs, The (1991)  3578: Gladiator (2000)  \
1.0                                      2                     3.0   
2.0                                      4                     3.0   
3.0                                      3                     0.0   
4.0                                      5                     4.0   
5.0                                      2                     2.0   

     1: Toy Story (1995)  2028: Saving Private Ryan (1998)  \
1.0                    1                                 2   
2.0                    8                                 2   
3.0                    2                                 3   
4.0                    5                                 2   
5.0                    1                                 2   

     296: Pulp Fiction (1994)  1259: Stand by Me (1986)  \
1.0                         1                         4   
2.0                         3                         1   
3.0                         3                         2   
4.0                         3                         3   
5.0                         1                         1   

     2396: Shakespeare in Love (1998)  2916: Total Recall (1990)  \
1.0                               2.0                        5.0   
2.0                               3.0                        4.0   
3.0                               3.0                        2.0   
4.0                               0.0                        1.0   
5.0                               3.0                        0.0   

     780: Independence Day (ID4) (1996)  541: Blade Runner (1982)  \
1.0                                   2                       0.0   
2.0                                   4                       3.0   
3.0                                   3                       2.0   
4.0                                   3                       3.0   
5.0                                   1                       1.0   

     1265: Groundhog Day (1993)  2571: Matrix, The (1999)  \
1.0                           2                         3   
2.0                           3                         3   
3.0                           2                         1   
4.0                           1                         3   
5.0                           4                         2   

     527: Schindler's List (1993)  2762: Sixth Sense, The (1999)  \
1.0                           2.0                              4   
2.0                           4.0                              1   
3.0                           2.0                              3   
4.0                           0.0                              1   
5.0                           4.0                              3   

     1198: Raiders of the Lost Ark (1981)  34: Babe (1995)  
1.0                                   3.0     

In [11]:
high_ranking_perc_df =(ranking_perc_df.loc[ranking_perc_df.index>=4.0].sum()/ranking_perc_df.sum()).sort_values(ascending=False)

In [12]:
high_ranking_perc_df

318: Shawshank Redemption, The (1994)                      0.700000
260: Star Wars: Episode IV - A New Hope (1977)             0.533333
3578: Gladiator (2000)                                     0.500000
541: Blade Runner (1982)                                   0.444444
593: Silence of the Lambs, The (1991)                      0.437500
2571: Matrix, The (1999)                                   0.416667
1265: Groundhog Day (1993)                                 0.416667
34: Babe (1995)                                            0.400000
296: Pulp Fiction (1994)                                   0.363636
2028: Saving Private Ryan (1998)                           0.363636
1259: Stand by Me (1986)                                   0.363636
1210: Star Wars: Episode VI - Return of the Jedi (1983)    0.357143
1: Toy Story (1995)                                        0.352941
527: Schindler's List (1993)                               0.333333
2762: Sixth Sense, The (1999)                   

#### Top movies for someone who has seen Toy Story: 
Calculate movies that most often occur with Movie #1: Toy Story, using the (x+y)/x method described in class. In other words, for each movie, calculate the percentage of Toy Story raters who also rated that movie. Order with the highest percentage first, and submit the top 3 (along with the correlations for the top two). Note, you will have ties -- to break the ties, use the lowest-numbered movie as the higher-ranked one. In other words, if Movies 541 and 318 are tied, then 318 gets the higher rank.

In [13]:
movies_toys_df = movies_df[movies_df['1: Toy Story (1995)'].notnull()].copy()

In [14]:
movies_toys_df

User  Gender (1 =F, 0=M)  260: Star Wars: Episode IV - A New Hope (1977)  \
0    755                   0                                             1.0   
1   5277                   0                                             5.0   
2   1577                   1                                             NaN   
3   4388                   0                                             NaN   
5   3823                   1                                             2.0   
7   5347                   0                                             4.0   
8   4117                   1                                             5.0   
9   2765                   0                                             4.0   
10  5450                   1                                             2.0   
11   139                   0                                             3.0   
12  1940                   0                                             2.0   
13  3118                   1                                             3.0   
14  4656                   1                                             4.0   
16  6037                   0                                             NaN   
17  3048                   1                                             4.0   
18  4790                   0                                             5.0   
19  4489                   0                                             1.0   

    1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
0                                                 5.0         
1                                                 3.0         
2                                                 NaN         
3                                                 3.0         
5                                                 4.0         
7                                                 NaN         
8                                                 1.0         
9                                                 2.0         
10                                                1.0         
11                                                5.0         
12                                                3.0         
13                                                NaN         
14                                                4.0         
16                                                NaN         
17                                                5.0         
18                                                1.0         
19                                                2.0         

    356: Forrest Gump (1994)  318: Shawshank Redemption, The (1994)  \
0                        2.0                                    NaN   
1                        NaN                                    2.0   
2                        NaN                                    5.0   
3                        NaN                                    NaN   
5                        4.0                                    4.0   
7                        NaN                                    NaN   
8                        NaN                                    4.0   
9                        NaN                                    5.0   
10                       5.0                                    NaN   
11                       2.0                                    NaN   
12                       NaN                                    5.0   
13                       3.0                                    NaN   
14                       NaN                                    NaN   
16                       NaN                                    NaN   
17                       1.0                                    5.0   
18                       NaN                                    NaN   
19                       2.0                                    4.0   

    593: Silence of the Lambs, The (1991)  3578: Gladiator (2000)  \
0                                     4.0                     4.0   
1          

In [15]:
ranking_toys_df =movies_toys_df.iloc[:,2:].T.count(axis=1)

In [16]:
ranking_toys_df

260: Star Wars: Episode IV - A New Hope (1977)             14
1210: Star Wars: Episode VI - Return of the Jedi (1983)    13
356: Forrest Gump (1994)                                    7
318: Shawshank Redemption, The (1994)                       8
593: Silence of the Lambs, The (1991)                      13
3578: Gladiator (2000)                                      9
1: Toy Story (1995)                                        17
2028: Saving Private Ryan (1998)                            8
296: Pulp Fiction (1994)                                   10
1259: Stand by Me (1986)                                    9
2396: Shakespeare in Love (1998)                            8
2916: Total Recall (1990)                                  11
780: Independence Day (ID4) (1996)                         13
541: Blade Runner (1982)                                    7
1265: Groundhog Day (1993)                                 11
2571: Matrix, The (1999)                                    9
527: Sch

In [17]:
((ranking_toys_df)/ranking_toys_df['1: Toy Story (1995)']).sort_values(ascending=False)

1: Toy Story (1995)                                        1.000000
260: Star Wars: Episode IV - A New Hope (1977)             0.823529
1210: Star Wars: Episode VI - Return of the Jedi (1983)    0.764706
593: Silence of the Lambs, The (1991)                      0.764706
780: Independence Day (ID4) (1996)                         0.764706
1265: Groundhog Day (1993)                                 0.647059
2916: Total Recall (1990)                                  0.647059
296: Pulp Fiction (1994)                                   0.588235
2762: Sixth Sense, The (1999)                              0.588235
527: Schindler's List (1993)                               0.588235
3578: Gladiator (2000)                                     0.529412
1259: Stand by Me (1986)                                   0.529412
1198: Raiders of the Lost Ark (1981)                       0.529412
2571: Matrix, The (1999)                                   0.529412
2028: Saving Private Ryan (1998)                

#### Correlation with Toy Story
Calculate the correlation between the vectors of ratings for Toy Story and each other movie. You can use the built-in CORREL() function. Order by the highest correlation (positive only) and submit the top 3 along with the correlation values for the top 2. Notice the differences between co-occurrence and correlation; these metrics are showing different types of relationships.

In [18]:
cols = set(movies_df.columns)
cols.remove('Gender (1 =F, 0=M)')
cols = list(cols)
movies_corr_df = movies_df[cols]

In [19]:
movies_corr_df.corr()['1: Toy Story (1995)'].sort_values(ascending=False)

1: Toy Story (1995)                                        1.000000
318: Shawshank Redemption, The (1994)                      0.888523
34: Babe (1995)                                            0.811107
296: Pulp Fiction (1994)                                   0.709842
2028: Saving Private Ryan (1998)                           0.596849
356: Forrest Gump (1994)                                   0.522913
541: Blade Runner (1982)                                   0.329634
3578: Gladiator (2000)                                     0.253396
2916: Total Recall (1990)                                  0.163796
2396: Shakespeare in Love (1998)                           0.101768
1265: Groundhog Day (1993)                                -0.062858
780: Independence Day (ID4) (1996)                        -0.069923
260: Star Wars: Episode IV - A New Hope (1977)            -0.119005
User                                                      -0.189503
527: Schindler's List (1993)                    

#### Mean rating difference by gender
First, recompute the mean rating for each movie separately for males and for females. And calculate the overall mean rating (across all ratings) for males and females. Submit the two movies that have the greatest differences (one where men are most above women, and one where women are most above men) along with the differences in average. Also submit the difference in overall rating averages (female average - male average).

In [41]:
movies_mf_mean_df = movies_df.iloc[:,1:].groupby('Gender (1 =F, 0=M)').mean()

In [42]:
movies_mf_mean_df

260: Star Wars: Episode IV - A New Hope (1977)  \
Gender (1 =F, 0=M)                                                   
0                                                         3.125000   
1                                                         3.428571   

                    1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
Gender (1 =F, 0=M)                                                            
0                                                                 3.0         
1                                                                 3.0         

                    356: Forrest Gump (1994)  \
Gender (1 =F, 0=M)                             
0                                       2.25   
1                                       3.00   

                    318: Shawshank Redemption, The (1994)  \
Gender (1 =F, 0=M)                                          
0                                                     3.4   
1                                                     3.8   

                    593: Silence of the Lambs, The (1991)  \
Gender (1 =F, 0=M)                                          
0                                                3.333333   
1                                                2.714286   

                    3578: Gladiator (2000)  1: Toy Story (1995)  \
Gender (1 =F, 0=M)                                                
0                                 2.833333             2.300000   
1                                 3.000000             3.571429   

                    2028: Saving Private Ryan (1998)  \
Gender (1 =F, 0=M)                                     
0                                           3.142857   
1                                           2.750000   

                    296: Pulp Fiction (1994)  1259: Stand by Me (1986)  \
Gender (1 =F, 0=M)                                                       
0                                      2.625                  3.000000   
1                                      4.000                  2.428571   

                    2396: Shakespeare in Love (1998)  \
Gender (1 =F, 0=M)                                     
0                                           2.142857   
1                                           4.250000   

                    2916: Total Recall (1990)  \
Gender (1 =F, 0=M)                              
0                                    2.200000   
1                                    1.714286   

                    780: Independence Day (ID4) (1996)  \
Gender (1 =F, 0=M)                                       
0                                             2.857143   
1                                             2.666667   

                    541: Blade Runner (1982)  1265: Groundhog Day (1993)  \
Gender (1 =F, 0=M)                                                         
0                                        3.0                    3.500000   
1                                        3.5                    2.833333   

                    2571: Matrix, The (1999)  527: Schindler's List (1993)  \
Gender (1 =F, 0=M)                                                           
0                                   3.142857                           3.5   
1                                   2.400000                           2.5   

                    2762: Sixth Sense, The (1999)  \
Gender (1 =F, 0=M)                                  
0                                        2.666667   
1                                        3.000000   

                    1198: Raiders of the Lost Ark (1981)  34: Babe (1995)  
Gender (1 =F, 0=M)                                                         
0                                               3.666667         2.000000  
1                                               2.000000         3.428571

In [45]:
diff_mf_df =movies_mf_mean_df.loc[1] - movies_mf_mean_df.loc[0]

In [47]:
diff_mf_df.sort_values(ascending=False)

2396: Shakespeare in Love (1998)                           2.107143
34: Babe (1995)                                            1.428571
296: Pulp Fiction (1994)                                   1.375000
1: Toy Story (1995)                                        1.271429
356: Forrest Gump (1994)                                   0.750000
541: Blade Runner (1982)                                   0.500000
318: Shawshank Redemption, The (1994)                      0.400000
2762: Sixth Sense, The (1999)                              0.333333
260: Star Wars: Episode IV - A New Hope (1977)             0.303571
3578: Gladiator (2000)                                     0.166667
1210: Star Wars: Episode VI - Return of the Jedi (1983)    0.000000
780: Independence Day (ID4) (1996)                        -0.190476
2028: Saving Private Ryan (1998)                          -0.392857
2916: Total Recall (1990)                                 -0.485714
1259: Stand by Me (1986)                        

In [49]:
movies_mf_mean_df.loc[1].mean() - movies_mf_mean_df.loc[0].mean()

0.11500000000000021

In [51]:
movies_mf_mean_df.loc[1].mean()

2.9992857142857146

In [50]:
movies_mf_mean_df.loc[0].mean()

2.8842857142857143

#### Next, compute the % of ratings 4+ separately for males and females. 
You'll be asked to submit two movies as above (largest difference in each direction). And again you'll indicate whether men or women are more likely to rate movies 4 stars or above.

In [60]:
titles = list(movies_df.columns.values)[2:]

In [57]:
titles

Index(['260: Star Wars: Episode IV - A New Hope (1977)',
       '1210: Star Wars: Episode VI - Return of the Jedi (1983)',
       '356: Forrest Gump (1994)', '318: Shawshank Redemption, The (1994)',
       '593: Silence of the Lambs, The (1991)', '3578: Gladiator (2000)',
       '1: Toy Story (1995)', '2028: Saving Private Ryan (1998)',
       '296: Pulp Fiction (1994)', '1259: Stand by Me (1986)',
       '2396: Shakespeare in Love (1998)', '2916: Total Recall (1990)',
       '780: Independence Day (ID4) (1996)', '541: Blade Runner (1982)',
       '1265: Groundhog Day (1993)', '2571: Matrix, The (1999)',
       '527: Schindler's List (1993)', '2762: Sixth Sense, The (1999)',
       '1198: Raiders of the Lost Ark (1981)', '34: Babe (1995)'],
      dtype='object')

In [107]:
gender_df = movies_df.iloc[:,1:].set_index('Gender (1 =F, 0=M)')

In [108]:
ratings_f_df = gender_df.loc[1] 
ratings_m_df = gender_df.loc[0] 

In [109]:
ratings_f_df

260: Star Wars: Episode IV - A New Hope (1977)  \
Gender (1 =F, 0=M)                                                   
1                                                              NaN   
1                                                              4.0   
1                                                              2.0   
1                                                              5.0   
1                                                              2.0   
1                                                              3.0   
1                                                              4.0   
1                                                              NaN   
1                                                              4.0   

                    1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
Gender (1 =F, 0=M)                                                            
1                                                                 NaN         
1                                                                 3.0         
1                                                                 4.0         
1                                                                 1.0         
1                                                                 1.0         
1                                                                 NaN         
1                                                                 4.0         
1                                                                 NaN         
1                                                                 5.0         

                    356: Forrest Gump (1994)  \
Gender (1 =F, 0=M)                             
1                                        NaN   
1                                        4.0   
1                                        4.0   
1                                        NaN   
1                                        5.0   
1                                        3.0   
1                                        NaN   
1                                        1.0   
1                                        1.0   

                    318: Shawshank Redemption, The (1994)  \
Gender (1 =F, 0=M)                                          
1                                                     5.0   
1                                                     1.0   
1                                                     4.0   
1                                                     4.0   
1                                                     NaN   
1                                                     NaN   
1                                                     NaN   
1                                                     NaN   
1                                                     5.0   

                    593: Silence of the Lambs, The (1991)  \
Gender (1 =F, 0=M)                                          
1                                                     2.0   
1                                                     4.0   
1                                                     NaN   
1                                                     2.0   
1                                                     NaN   
1                                                     2.0   
1                                                     5.0   
1                                                     3.0   
1                                                     1.0   

                    3578: Gladiator (2000)  1: Toy Story (1995)  \
Gender (1 =F, 0=M)                                                
1                                      NaN                  4.0   
1                                      1.0                  NaN   
1                                      NaN                  3.0   
1                                      4.0                  4.0   
1                                      5.0                  5.0   
1                                    

In [110]:
ranking_perc_f_df = ratings_f_df.apply(lambda x: x.value_counts()).fillna(0)
high_ranking_perc_f_df =(ranking_perc_f_df.loc[ranking_perc_f_df.index>=4.0].sum()/
                       ranking_perc_f_df.sum()).sort_values(ascending=False)
ranking_perc_m_df = ratings_m_df.apply(lambda x: x.value_counts()).fillna(0)
high_ranking_perc_m_df =(ranking_perc_m_df.loc[ranking_perc_m_df.index>=4.0].sum()/
                       ranking_perc_m_df.sum()).sort_values(ascending=False)

In [111]:
merge_high_ranking_perc_df = pd.concat([high_ranking_perc_m_df, high_ranking_perc_f_df], axis=1, join='inner')

In [112]:
merge_high_ranking_perc_df.columns = ['M','F']

In [113]:
merge_high_ranking_perc_df['diff']=(merge_high_ranking_perc_df['F']-merge_high_ranking_perc_df['M'])
merge_high_ranking_perc_df

M         F  \
318: Shawshank Redemption, The (1994)               0.600000  0.800000   
2571: Matrix, The (1999)                            0.571429  0.200000   
593: Silence of the Lambs, The (1991)               0.555556  0.285714   
260: Star Wars: Episode IV - A New Hope (1977)      0.500000  0.571429   
527: Schindler's List (1993)                        0.500000  0.166667   
1265: Groundhog Day (1993)                          0.500000  0.333333   
1198: Raiders of the Lost Ark (1981)                0.500000  0.000000   
3578: Gladiator (2000)                              0.500000  0.500000   
2762: Sixth Sense, The (1999)                       0.333333  0.333333   
2028: Saving Private Ryan (1998)                    0.285714  0.500000   
780: Independence Day (ID4) (1996)                  0.285714  0.333333   
1210: Star Wars: Episode VI - Return of the Jed...  0.250000  0.500000   
1259: Stand by Me (1986)                            0.250000  0.428571   
296: Pulp Fiction (1994)                            0.250000  0.666667   
1: Toy Story (1995)                                 0.200000  0.571429   
2916: Total Recall (1990)                           0.200000  0.000000   
541: Blade Runner (1982)                            0.200000  0.750000   
2396: Shakespeare in Love (1998)                    0.000000  0.750000   
356: Forrest Gump (1994)                            0.000000  0.500000   
34: Babe (1995)                                     0.000000  0.571429   

                                                        diff  
318: Shawshank Redemption, The (1994)               0.200000  
2571: Matrix, The (1999)                           -0.371429  
593: Silence of the Lambs, The (1991)              -0.269841  
260: Star Wars: Episode IV - A New Hope (1977)      0.071429  
527: Schindler's List (1993)                       -0.333333  
1265: Groundhog Day (1993)                         -0.166667  
1198: Raiders of the Lost Ark (1981)               -0.500000  
3578: Gladiator (2000)                              0.000000  
2762: Sixth Sense, The (1999)                       0.000000  
2028: Saving Private Ryan (1998)                    0.214286  
780: Independence Day (ID4) (1996)                  0.047619  
1210: Star Wars: Episode VI - Return of the Jed...  0.250000  
1259: Stand by Me (1986)                            0.178571  
296: Pulp Fiction (1994)                            0.416667  
1: Toy Story (1995)                                 0.371429  
2916: Total Recall (1990)                          -0.200000  
541: Blade Runner (1982)                            0.550000  
2396: Shakespeare in Love (1998)                    0.750000  
356: Forrest Gump (1994)                            0.500000  
34: Babe (1995)                                     0.571429

In [114]:
merge_high_ranking_perc_df.sort_values('diff',ascending=False)

M         F  \
2396: Shakespeare in Love (1998)                    0.000000  0.750000   
34: Babe (1995)                                     0.000000  0.571429   
541: Blade Runner (1982)                            0.200000  0.750000   
356: Forrest Gump (1994)                            0.000000  0.500000   
296: Pulp Fiction (1994)                            0.250000  0.666667   
1: Toy Story (1995)                                 0.200000  0.571429   
1210: Star Wars: Episode VI - Return of the Jed...  0.250000  0.500000   
2028: Saving Private Ryan (1998)                    0.285714  0.500000   
318: Shawshank Redemption, The (1994)               0.600000  0.800000   
1259: Stand by Me (1986)                            0.250000  0.428571   
260: Star Wars: Episode IV - A New Hope (1977)      0.500000  0.571429   
780: Independence Day (ID4) (1996)                  0.285714  0.333333   
2762: Sixth Sense, The (1999)                       0.333333  0.333333   
3578: Gladiator (2000)                              0.500000  0.500000   
1265: Groundhog Day (1993)                          0.500000  0.333333   
2916: Total Recall (1990)                           0.200000  0.000000   
593: Silence of the Lambs, The (1991)               0.555556  0.285714   
527: Schindler's List (1993)                        0.500000  0.166667   
2571: Matrix, The (1999)                            0.571429  0.200000   
1198: Raiders of the Lost Ark (1981)                0.500000  0.000000   

                                                        diff  
2396: Shakespeare in Love (1998)                    0.750000  
34: Babe (1995)                                     0.571429  
541: Blade Runner (1982)                            0.550000  
356: Forrest Gump (1994)                            0.500000  
296: Pulp Fiction (1994)                            0.416667  
1: Toy Story (1995)                                 0.371429  
1210: Star Wars: Episode VI - Return of the Jed...  0.250000  
2028: Saving Private Ryan (1998)                    0.214286  
318: Shawshank Redemption, The (1994)               0.200000  
1259: Stand by Me (1986)                            0.178571  
260: Star Wars: Episode IV - A New Hope (1977)      0.071429  
780: Independence Day (ID4) (1996)                  0.047619  
2762: Sixth Sense, The (1999)                       0.000000  
3578: Gladiator (2000)                              0.000000  
1265: Groundhog Day (1993)                         -0.166667  
2916: Total Recall (1990)                          -0.200000  
593: Silence of the Lambs, The (1991)              -0.269841  
527: Schindler's List (1993)                       -0.333333  
2571: Matrix, The (1999)                           -0.371429  
1198: Raiders of the Lost Ark (1981)               -0.500000